pkg/operator/ceph/object/admin.go

```diff
@@ -33,17 +33,24 @@ import (
 
 // Context holds the context for the object store.
 type Context struct {
-	Context               *clusterd.Context
-	clusterInfo           *cephclient.ClusterInfo
-	Name                  string
-	UID                   string
-	Endpoint              string
-	Realm                 string
-	ZoneGroup             string
-	Zone                  string
-	adminOpsUserAccessKey string
-	adminOpsUserSecretKey string
-	adminOpsClient        *admin.API
+	Context     *clusterd.Context
+	clusterInfo *cephclient.ClusterInfo
+	Name        string
+	UID         string
+	Endpoint    string
+	Realm       string
+	ZoneGroup   string
+	Zone        string
+}
+
+// AdminOpsContext holds the object store context as well as information for connecting to the admin
+// ops API.
+type AdminOpsContext struct {
+	Context
+	TlsCert               []byte
+	AdminOpsUserAccessKey string
+	AdminOpsUserSecretKey string
+	AdminOpsClient        *admin.API
 }
```

Comment:
Separated these two contexts because it is very confusing to know when a reused context can and cannot be expected to have certain fields filled in. By separating them, it is clear when the admin ops info is needed for a function and when it isn't.